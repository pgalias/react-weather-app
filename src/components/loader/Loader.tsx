import React, { FunctionComponent } from 'react';
import ReactAnimatedWeather from 'react-animated-weather';
import styles from './loader.module.css';

export const Loader: FunctionComponent = () => (
  <div className={styles.loaderContainer} role="alert" aria-busy="true">
    <span className={`${styles.loader} animate-spin`}>
      <ReactAnimatedWeather icon="CLEAR_DAY" size={128} animate={false} color="#D97706" />
    </span>
  </div>
);
