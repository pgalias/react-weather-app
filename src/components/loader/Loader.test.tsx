import React from 'react';
import { render, screen } from '@testing-library/react';
import { Loader } from './Loader';

jest.mock('react-animated-weather', () => () => <span>loading icon</span>);

describe('Loader component', () => {
  test('should match snapshot', () => {
    render(<Loader />);

    expect(screen.getByRole('alert')).toMatchSnapshot();
  });
});
