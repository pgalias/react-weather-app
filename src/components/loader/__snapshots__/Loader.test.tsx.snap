// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should match snapshot 1`] = `
<div
  aria-busy="true"
  class="loaderContainer"
  role="alert"
>
  <span
    class="loader animate-spin"
  >
    <span>
      loading icon
    </span>
  </span>
</div>
`;
